openlibrary/templates/lib/header_dropdown.html

```diff
@@ -12,25 +12,52 @@
 <div class="$(props['name'])-component header-dropdown">
     <details>
       <summary>
-        $if 'image' in props:
+        $if 'mask' in props and ctx.user:
+            <img class="account__icon" src="https://archive.org/services/img/$(get_internet_archive_id(ctx.user.key))" alt="My account">
+            <img class="header-dropdown__icon logged" src="$(props['image'])" alt="$(props['label'])"/>
+        $elif 'image' in props:
             <img class="header-dropdown__icon" src="$(props['image'])" alt="$(props['label'])"/>
         $elif 'label' in props:
             $(props['label'])
             <span class="shift">$_('Menu')</span>
 
         <img class="down-arrow" aria-hidden="true" src="/static/images/down-arrow.png" alt="" role="presentation" width="7" height="4">
       </summary>
-      <div class="$(props['name'])-dropdown-component navigation-dropdown-component">
+      $if 'mask' in props:
+        <div class="mask-menu"></div>
+      <div 
+      $if 'mask' in props:
```

Comment:
Ah, don't use mask a proxy for checking if it's the hamburger icon. That _very_ unclear. Add an `'is_hamburger_component'` property for now, until we do the larger refactor in a future PR.

Reply:
Actually, use `props['name'] = 'hamburger'`, like you do elsewhere.